src/types.ts

```diff
@@ -23,6 +24,12 @@ export type AnyState = State<any, any>;
 
 export type SourceProvider = 'gist' | 'registry';
 
+export interface SourceRegistryData extends SourceFileFragment {
+  // we can't trust SSR data to be accurate because at the moment we can use authenticated user during SSR
```

Comment:
Why can't we use authenticated user info during SSR? If we have access to the cookie? Or do you mean that we don't.

Reply:
We don't have access to the cookie because Supabase keeps the token in the `localStorage`. I had a cookie-based version (using some hacks~) locally but that would mean that we would be keeping the refresh token in more than one place (security concern) or that we wouldn't be able to refresh tokens during SSR (back to square one - some requests couldn't be reliably SSRed).

Reply:
btw. I have not ended up using the ssr/client distinction - at first I thought I've though that I will need it but it turned out not to be true. I've decided to keep this property though as it seems to have some informative value